cmd/cmd.go

```diff
@@ -47,17 +53,61 @@ func CreateHandler(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	var spinner *Spinner
+	modelfile, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer modelfile.Close()
+
+	commands, err := parser.Parse(modelfile)
+	if err != nil {
+		return err
+	}
+
+	spinner := NewSpinner("transferring context")
+	go spinner.Spin(100 * time.Millisecond)
+
+	for i := range commands {
+		switch commands[i].Name {
+		case parser.CommandFrom, parser.CommandAdapter:
+			binfile, err := os.Open(commands[i].Args)
+			switch {
+			case errors.Is(err, os.ErrNotExist) && commands[i].Name == parser.CommandFrom:
+				// model might be a model name and not a file path
+				continue
+			case err != nil:
+				return err
+			}
+			defer binfile.Close()
+
+			hash := sha256.New()
+			if _, err := io.Copy(hash, binfile); err != nil {
```

Comment:
We should have some status update for the spinner here because this could take a really long time

Reply:
Parsing the modelfile falls under `transferring context` which is already a status. I don't feel a need to update each individual step